### PR TITLE
Including <uv.h> shouldn't emit any warnings

### PR DESCRIPTION
**Pull request by trondn**
I'm trying to add a plugin that use libuv to my library and I'm normally compiling my modules with:

 gcc -DHAVE_CONFIG_H -I. -I./src -pipe -I./include -DHAVE_VISIBILITY=1 -fvisibility=hidden -Wall -pedantic -Wundef -Wshadow -fdiagnostics-show-option -Wformat -fno-strict-aliasing -Wno-strict-aliasing -Wextra -Werror -fprofile-arcs -ftest-coverage -pipe -std=gnu99 -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wmissing-declarations -Wcast-align 

The constructs:
# if variable

will emit a warning if it's not defined, causing my build to break. The attached patch change these to #ifdef (this means that defining them to "0" instead of undef'ing them no longer works.. The alternative would be to extend the code check if they're defined _AND_ set to something else than 0...

### CODE REVIEW

File: include/uv.h [hunk 1]
```diff
@@ -53,7 +53,7 @@
 #include <stdint.h> /* int64_t */
 #include <sys/types.h> /* size_t */
 
-#if defined(__unix__) || defined(__POSIX__) || defined(__APPLE__)
+#if defined(__unix__) || defined(__POSIX__) || defined(__APPLE__) || defined(__SVR4)
```
Comment (bnoordhuis) [inline include/uv.h]: Wrap at 80 chars.

**Comment by bnoordhuis** [comment]
I don't really want to pull in this patch. `#if FOO` has been used since the dawn of time.

**Comment by piscisaureus** [comment]
@bnoordhuis I don't see much harm in it. However, I am surprised that it triggers a warning. It never did for me (and I tried many platforms).

**Comment by bnoordhuis** [comment]
It only happens in gcc's overzealous mode, -Wundef, disabled by default because of that 'since the dawn of time' thing.

**Comment by trondn** [comment]
Given that this is a "public" header for the library it would be nice if it didn't generate warnings causing building of clients of the library to fail.. I can fix the line wrapping, but if the patches won't be accepted I'd rather just skip adding support for libuv for libcouchbase given that it'll require me to keep me own branch of libuv in order to make it compile (making upgrading stuff etc harder).

**Comment by ingenthr** [comment]
Since uv.h is something you want others to consume, and since others may want 'overzealous mode' for their projects, it seems like there's a good argument for including this change.  What's the downside?  It doesn't seem less readable to me.

**Comment by bnoordhuis** [comment]
Trond, can you sign [the CLA](http://nodejs.org/cla.html)?

**Comment by trondn** [comment]
I've just signed the CLA and submitted it on email. Please let me know if there is anything else I need to do.

**Comment by bnoordhuis** [comment]
@isaacs Where do emailed CLAs end up? It's not in the list.

**Comment by isaacs** [comment]
@trondn Where did you email the CLA?

**Comment by isaacs** [comment]
@trondn Nevermind.  Yes.  Confirmed.

@bnoordhuis I'll add it to the spreadsheet.  @trondn is good.  Proceed as normal.

**Comment by bnoordhuis** [comment]
Thanks Trond, landed in cbb9300 and 5143d54.

**Pull request closed without merging**